main.go

```diff
@@ -502,7 +507,26 @@ func (c *controller) sync() {
 			continue
 		}
 
+		// If there's an increase in replicas we poll for the new replicas to be ready
+		if _, ok := c.replicas[hashring]; ok && c.replicas[hashring] < *sts.Spec.Replicas {
+			// Iterate over new replicas to wait until they are running
+			for i := c.replicas[hashring]; i < *sts.Spec.Replicas; i++ {
+				start := time.Now()
+				podName := fmt.Sprintf("%s-%d", sts.Name, i)
+
+				if err := c.waitForPod(podName); err != nil {
+					level.Warn(c.logger).Log("msg", "failed polling until pod is ready", "pod", podName, "duration", time.Since(start), "err", err)
+					continue
```

Comment:
why do we continue here if failed polling? Shuoldn't we throw the error instead?